accord-core/src/main/java/accord/coordinate/CollectLatestDeps.java

```diff
@@ -95,10 +109,15 @@ public void onFailure(Id from, Throwable failure)
         if (isDone)
             return;
 
+        if (failure != null)
+            this.failure = FailureAccumulator.append(this.failure, failure);
+
         if (tracker.recordFailure(from) == Failed)
         {
             isDone = true;
-            callback.accept(null, new Timeout(txnId, route.homeKey()));
+            if (this.failure == null)
```

Comment:
looks like above we can report Failure as a replay, so it not necessarily is a timeout, or?

Reply:
We don't really distinguish between Timeout and having exhausted our options in most places. We do technically have `Exhausted`, I can switch to that if you like? It is probably more correct. We should at some point go through all our exception logic and make sure it is consistent, I noticed already a number of inconsistencies.

Reply:
Maybe add a TODO for now and we can do all of them in one sweep?

Reply:
I've just updated to Exhausted for now. There's already TODOs littered about the place to look at our exception handling, but I'll add another here as well to be sure.